source/modulo_core/src/communication/PublisherInterface.cpp

```diff
@@ -18,69 +18,13 @@ PublisherInterface::PublisherInterface(PublisherType type, std::shared_ptr<Messa
     : type_(type), message_pair_(std::move(message_pair)) {}
 
 void PublisherInterface::activate() {
-  if (this->message_pair_ == nullptr) {
-    throw exceptions::NullPointerException("Message pair is not set, cannot deduce message type");
-  }
-  switch (this->message_pair_->get_type()) {
-    case MessageType::BOOL:
-      this->template get_handler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Bool>, std_msgs::msg::Bool>()
-          ->on_activate();
-      break;
-    case MessageType::FLOAT64:
-      this->template get_handler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Float64>, std_msgs::msg::Float64>()
-          ->on_activate();
-      break;
-    case MessageType::FLOAT64_MULTI_ARRAY:
-      this->template get_handler<
-              rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Float64MultiArray>,
-              std_msgs::msg::Float64MultiArray>()
-          ->on_activate();
-      break;
-    case MessageType::INT32:
-      this->template get_handler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Int32>, std_msgs::msg::Int32>()
-          ->on_activate();
-      break;
-    case MessageType::STRING:
-      this->template get_handler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::String>, std_msgs::msg::String>()
-          ->on_activate();
-      break;
-    case MessageType::ENCODED_STATE:
-      this->template get_handler<rclcpp_lifecycle::LifecyclePublisher<EncodedState>, EncodedState>()->on_activate();
-      break;
-  }
+  throw exceptions::CoreException(
+      "The publisher handler used is responsible for handling publisher activation, but it doesn't do so!");
```

Comment:
```suggestion
      "The derived publisher handler is required to override this function to handle activation");
```
Similar in deactivation